doc/rtd/reference/merging.rst

```diff
@@ -3,114 +3,179 @@
 
-- :command:`Dict`
-- :command:`List`
-- :command:`String`
+Available options
+-----------------
 
-``Dict``
---------
+Objects
+~~~~~~~
 
-The :command:`Dict` merger has the following options, which control what is
-done with values contained within the config.
+These options apply to objects:
 
 - :command:`allow_delete`: Existing values not present in the new value can be
-  deleted. Defaults to ``False``.
+  deleted. Disabled by default.
 - :command:`no_replace`: Do not replace an existing value if one is already
   present. Enabled by default.
-- :command:`replace`: Overwrite existing values with new ones.
+- :command:`replace`: Overwrite existing values with new ones. Disabled by 
+  default.
 
-``List``
---------
+Lists
+~~~~~
 
-The :command:`List` merger has the following options, which control what is
-done with the values contained within the config.
+These options apply to lists:
 
-- :command:`append`: Add new value to the end of the list. Defaults to
-  ``False``.
-- :command:`prepend`: Add new values to the start of the list. Defaults to
-  ``False``.
+- :command:`append`: Add new value to the end of the list. Disabled by 
+  default.
+- :command:`prepend`: Add new values to the start of the list. Disabled by
+  default.
 - :command:`no_replace`: Do not replace an existing value if one is already
   present. Enabled by default.
-- :command:`replace`: Overwrite existing values with new ones.
+- :command:`replace`: Overwrite existing values with new ones. Disabled by
+  default.
 
-String
-------
+Strings
+~~~~~~~
 
-The :command:`Str` merger has the following options, which control what is
-done with the values contained within the config.
+These options apply to strings:
 
-- :command:`append`: Add new value to the end of the string. Defaults to
-  False.
+- :command:`append`: Add new value to the end of the string. Disabled by 
+  default.
 
 Common options
---------------
+~~~~~~~~~~~~~~
 
 These are the common options for all merge types, which control how recursive
 merging is done on other types.
 
-- :command:`recurse_dict`: If ``True``, merge the new values of the
-  dictionary. Defaults to ``True``.
-- :command:`recurse_list`: If ``True``, merge the new values of the list.
-  Defaults to ``False``.
+- :command:`recurse_dict`: Merge the new values of the dictionary. Enabled by
+  default.
+- :command:`recurse_list`: Merge the new values of the list. Disabled by 
+  default.
 - :command:`recurse_array`: Alias for ``recurse_list``.
-- :command:`recurse_str`: If ``True``, merge the new values of the string.
-  Defaults to False.
+- :command:`recurse_str`: Merge the new values of the string. Disabled by 
+  default.
+
+Example cloud-config
+====================
+
+A common request is to include multiple ``runcmd`` directives in different
+files and merge all of the commands together. To achieve this, we must modify
+the default merging to allow for dictionaries to join list values.
+
+The first config:
```

Comment:
We don't need the merge config in both configs. The first one encountered will apply until a new one is encountered.

Reply:
That's something I would like to add to the page, I don't think that's particularly obvious yet, but good to know!

Reply:
I just realised that's already present in the "Specifying multiple types" section, although I still think that behaviour is really confusing and not explained particularly well.  
If I understood this correctly, the set of merging strategies that apply to a given file will be derived from partially applying them to the previous set, where the initial set will be the default. That is:

|  # | effective strategy                      | origin        | definition            |
|---:|:----------------------------------------|---------------|:----------------------|
|  0 | `dict()+list()+str()`                   | _default_     | `dict()+list()+str()` |
|  1 | `dict()+list(append)+str()`             | config-1.yaml | `list(append)`        |
|  2 | `dict(recurse_list)+list(append)+str()` | config-2.yaml | `dict(recurse_list)`  |

If this assumption is correct, it feels like having to specify a bit mask that will be OR-ed with another mask I possibly don't know the value of: What if the previous strategy set strings to append without my knowledge? That means the only sane thing to do would specify the merge strategy as defensively as possible to prevent errors from cropping up.  
This feels worth pointing out, what do you think?

Reply:
Yes, I agree. This is a good point and worth pointing out.